orttraining/orttraining/core/session/training_session.cc

```diff
@@ -1038,6 +1102,88 @@ Status TrainingSession::SetStateTensors(const NameMLValMap& state_tensors, bool
   return Status::OK();
 }
 
+Status TrainingSession::SetModelOptState(const std::unordered_map<std::string, NameMLValMap>& model_tensors, const std::unordered_map<std::string, NameMLValMap>& optimizer_tensors, bool strict) {
+  ORT_RETURN_IF_NOT(IsInitialized(), "Can't update initializers before session has been initialized.");
+
+  std::unordered_set<std::string> ckpt_initializer_names;
+  std::transform(model_tensors.begin(), model_tensors.end(),
+                 std::inserter(ckpt_initializer_names, ckpt_initializer_names.end()),
+                 [](auto pair) { return pair.first; });
+
+  NameMLValMap initializers;
+  ORT_RETURN_IF_ERROR(GetSessionState().GetInitializedTensors(ckpt_initializer_names, !strict, initializers));
+
+  const std::unordered_set<std::string> valid_state_tensor_names = GetStateTensorNames();
+
+  for (const auto& fp_or_mp_tensors : model_tensors) {
+    for (const auto& state : fp_or_mp_tensors.second) {
+      const bool is_valid_state_tensor =
+          valid_state_tensor_names.find(state.first) != valid_state_tensor_names.end();
+      const auto initializer_it = initializers.find(state.first);
```

Comment:
This does not account for the graph having modified names but the input checkpoint state having original names.

Reply:
Sync offline, this will be moved to next PR